components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java

```diff
@@ -761,6 +758,9 @@ private void populateAuthnReqDTO(HttpServletRequest req, SAMLSSOAuthnReqDTO auth
         authnReqDTO.setClaimMapping(authResult.getClaimMapping());
         authnReqDTO.setTenantDomain(sessionDTO.getTenantDomain());
         authnReqDTO.setIdPInitSLOEnabled(sessionDTO.isIdPInitSLO());
+        if (StringUtils.isNotBlank(sessionDTO.getAttributeConsumingServiceIndex())) {
```

Comment:
What happens if this is a string but not a number ? Isn't the correct way to do this be checing StringUtils.isNumeric ?